ambry-messageformat/src/test/java/com/github/ambry/messageformat/BlobStoreHardDeleteTest.java

```diff
@@ -353,6 +355,34 @@ public void blobStoreHardDeleteTestBlobV2Simple() throws MessageFormatException,
     blobStoreHardDeleteTestUtil(blobVersions, blobTypes);
   }
 
+  @Test
+  public void blobStoreHardDeleteTestBlobV3Simple() throws MessageFormatException, IOException {
+    boolean useBlobVersionSaved = PutMessageFormatInputStream.useBlobFormatV3;
+    PutMessageFormatInputStream.useBlobFormatV3 = true;
+    try {
+      Consumer<BlobType> runTest = blobType -> {
+          short[] blobVersions = new short[5];
+          BlobType[] blobTypes = new BlobType[5];
+          for (int i = 0; i < 5; i++) {
+            blobVersions[i] = MessageFormatRecord.Blob_Version_V3;
+            blobTypes[i] = blobType;
+          }
+          try {
+            // all blobs V3 with Data blob
```

Comment:
this comment is probably copy-paste err since u r passing metadata blob below L380

Reply:
cleaned.